coucharchive

```diff
@@ -76,6 +77,30 @@ def _check_couchdb_connection(url):
     return auth_url
 
 
+def _get_couch_system_stats(url):
+    url = url + '/_node/_local/_system'
+    parts = list(urlsplit(url))
+    server = parts[1]
+    username, password = None, None
+    if '@' in server:
+        credentials, server = server.rsplit('@', 1)
+        credentials = credentials.split(':', 1)
+        username = credentials.pop(0)
+        password = credentials.pop(0) if credentials else None
+    parts[1] = server
+    url = urlunsplit(parts)       # http://server/db/
+    parts[1] = '%s:%s@%s' % (quote(username, safe=[]),
+                             quote(password, safe=[]),
+                             server)
+    auth = base64.b64encode(('%s:%s' % (username, password))
+                            .encode('utf-8')).decode('utf-8')
+    req = urllib.request.Request(url, headers={
+        'Authorization': 'Basic %s' % auth})
+    response = urllib.request.urlopen(req)
+
+    return json.loads(response.read().decode('utf-8'))
```

Comment:
To prevent code duplication, I propose to reuse `_couchdb_request()` from https://github.com/adrienverge/coucharchive/pull/20, would that be OK for you?

Reply:
It have been perfect thanks :-)